administration/src/components/generation/CardCreationModel.ts

```diff
@@ -6,15 +6,15 @@ export interface CardCreationModel {
     id: number,
     forename: string,
     surname: string,
-    expirationDate: Date,
+    expirationDate: Date | null,
     cardType: CardType
 }
 
 export const isNameValid = (value: string) => value.length > 0 && value.length < MAX_NAME_LENGTH
 
-export const isExpirationDateValid = (value: Date) => value > new Date()
+export const isExpirationDateValid = (value: Date | null) => value !== null && value > new Date()
 
 export const isValid = (cardCreationModel: CardCreationModel) =>
     isNameValid(cardCreationModel.forename)
     && isNameValid(cardCreationModel.surname)
-    && isExpirationDateValid(cardCreationModel.expirationDate)
+    && (isExpirationDateValid(cardCreationModel.expirationDate) || cardCreationModel.cardType === CardType.gold)
```

Comment:
I think we don't need the `cardType` check, do we? If we set CardType to gold, then we should also set expiration date to null automatically (already in the input form)